### PR TITLE
CMake cleanups for Windows

### PR DESCRIPTION
**Pull request by Orphis**

### CODE REVIEW

File: Externals/libusb/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,9 @@ add_library(usb SHARED EXCLUDE_FROM_ALL
     libusb/sync.c
 )
 set_target_properties(usb PROPERTIES VERSION 1.0.19)
+if(WIN32)
+    target_include_directories(usb BEFORE PUBLIC libusb PRIVATE msvc)
```
Comment (iwubcode) [inline Externals/libusb/CMakeLists.txt]: Just curious, why do we need the before statement?

Comment (Orphis) [inline Externals/libusb/CMakeLists.txt]: It is to include a "config.h". That's too generic, so putting it BEFORE the other include directories is way safer!
We currently add all the include directories for all the external libraries in the global scope and it's propagated to all of them. Until we remove all the global options and move to direct dependencies, we'll need things like that.

**Comment by iwubcode** [comment]
So just a general comment and I'll admit, I know little of the open source world.  But I've seen at least a project or two handle externals differently by using CMake to pull in the external code.  Is there a reason we don't do that?  Instead, we're writing our own building logic.  Do these external projects not use CMake?

**Comment by Orphis** [comment]
It's a tradeoff. By rebuilding a library directly, you simplify the process for building for new platforms (or current ones).
You don't have also build flags properly propagated easily by using prebuilt binaries or their build scripts.
Those libraries are usually not too difficult to build, so it's ok to do so.

**Pull request merged**